jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/api/Stream.java

```diff
@@ -298,4 +316,62 @@ public boolean onIdleTimeout(Stream stream, Throwable x)
             }
         }
     }
+
+    /**
+     * <p>An ordered list of frames belonging to the same stream.</p>
+     */
+    public static class FrameList
```

Comment:
why  a class rather than just some utility methods to build `List<StreamFrame>` ?  They could equally check that all frames are the same streamId, but ultimately why check at creation when you must have to check when sent.  If a single DataFrame is sent with the wrong streamId, it is the same problem

Reply:
I find using a dedicated class more elegant as it is a clear API indicator that a method accepting a `FrameList` is won't modify the collection of `StreamFrame`s. I'd just explicitly state in the javadoc of this class that immutable.